lib/meta.nix

```
@@ -139,18 +140,34 @@ rec {
        lib.getLicenseFromSpdxId "mIt" == lib.licenses.mit
        => true
        lib.getLicenseFromSpdxId "MY LICENSE"
-       => trace: warning: getLicenseFromSpdxId: No license matches the given SPDX ID: MY LICENSE
        => { shortName = "MY LICENSE"; }
   */
-  getLicenseFromSpdxId =
+  getLicenseFromSpdxId = licstr:
+    getLicenseFromSpdxId' licstr { shortName = licstr; };
+
+  /* Get the corresponding attribute in lib.licenses
+     from the SPDX ID,
+     or fallback to the given default value.
+
+     Type:
+       getLicenseFromSpdxId' :: str -> Any -> Any
+
+     Example:
+     lib.getLicenseFromSpdxId' "MIT" null == lib.licenses.mit
+     => true
+     lib.getLicenseFromSpdxId' "MY LICENSE" lib.licenses.free == lib.licenses.free
+     => true
+     lib.getLicenseFromSpdxId' "MY LICENSE" null
+     => null
+     lib.getLicenseFromSpdxId' "MY LICENSE" (builtins.abort "No SPDX ID matches MY LICENSE")
+     => error: evaluation aborted with the following error message: 'No SPDX ID matches MY LICENSE'
+  */
+  getLicenseFromSpdxId' =
```

```suggestion
  getLicenseFromSpdxIdOr =
```

This is then similar to `spdxLicenses.${ ... } or default`

---

I'd say let's just introduce the -`Or` variant for now. One step at a time :)

---

> I'd say let's just introduce the -Or variant for now. One step at a time :)

Addressed.